lock-api/src/main/java/com/palantir/lock/client/timestampleases/TimestampLeaseAcquirerImpl.java

```diff
@@ -16,28 +16,203 @@
     }
 
     @Override
     public TimestampLeaseResults acquireNamedTimestampLeases(Map<TimestampLeaseName, Integer> requests) {
-        // TODO(aalouane): implement
-        throw new UnsupportedOperationException("Not implemented yet");
+        TimestampLeaseResponses response = acquireLeasesWithRetry(requests).response();
+        try {
+            return TimestampLeaseResults.builder()
+                    .lock(createLeasedLockToken(response))
+                    .results(createTimestampLeaseResult(requests, response.getTimestampLeaseResponses()))
+                    .build();
+        } catch (RuntimeException | Error e) {
+            log.error("Unexpected exception while creating client results", e);
+            asyncUnlock(response);
+            throw e;
+        }
     }
 
     @Override
     public void close() {
         // TODO(aalouane): decide whether or not to close the unlocker depending on ownership
     }
+
+    private RequestAndResponse acquireLeasesWithRetry(Map<TimestampLeaseName, Integer> requests) {
+        try {
+            return retryer.call(() -> acquireLeases(requests));
+        } catch (ExecutionException e) {
+            // should not happen with how the retryer is set up
+            log.warn("Unexpected exception. Expected a retry exception", e);
+            throw new SafeRuntimeException(e);
+        } catch (RetryException e) {
+            Attempt<?> lastFailedAttempt = e.getLastFailedAttempt();
+            log.warn(
+                    "Exhausted retries to get enough timestamps while acquiring timestamp lease",
+                    SafeArg.of("requests", requests),
+                    SafeArg.of("numRetries", lastFailedAttempt.getAttemptNumber()),
+                    SafeArg.of("attemptHadException", lastFailedAttempt.hasException()),
+                    e);
+            throw new AtlasDbDependencyException(
+                    "Exhausted retries to get enough timestamps while acquiring timestamp lease",
+                    lastFailedAttempt.hasException() ? lastFailedAttempt.getExceptionCause() : null,
+                    SafeArg.of("requests", requests),
+                    SafeArg.of("numRetries", lastFailedAttempt.getAttemptNumber()),
+                    SafeArg.of(
+                            "maybeResult",
+                            lastFailedAttempt.hasResult() ? lastFailedAttempt.getResult() : Optional.empty()));
+        }
+    }
+
+    private RequestAndResponse acquireLeases(Map<TimestampLeaseName, Integer> requestMap) {
+        // we prefer to use a new request id for non-dialogue-native dialogue attempts
+        TimestampLeaseRequests requests = TimestampLeaseRequests.of(RequestId.of(uuidSupplier.get()), requestMap);
+
+        NamespaceTimestampLeaseRequest request = NamespaceTimestampLeaseRequest.of(List.of(requests));
+        List<TimestampLeaseResponses> response =
+                delegate.acquireTimestampLeases(request).getAlias();
```

Comment:
Wait, why did we do:

```
      NamespaceTimestampLeaseResponse:
         fields:
           alias: list<TimestampLeaseResponses>
```

here?

Reply:
Can you fix this up in a separate PR?

Reply:
I'll handle this

Reply:
Actually, I had a look and you can just do this inline in this PR, it's fine

Reply:
there's no other code that depends on this, so shouldn't make the diff larger

Reply:
The conjure codegen for single field value and alias is the same :)

Reply:
Fixing